avc/src/planner/planner.cpp

```diff
@@ -126,39 +177,107 @@ void planner::mapCb(const sensor_msgs::PointCloud2ConstPtr& map) {
     pcl::KdTreeFLANN<pcl::PointXYZ> kdtree;
     kdtree.setInputCloud(cloud);
 
-    //build paths and evaluate their weights
-    vector<double> weightedSteeringRaw;
-    double weightTotal = 0, cost, angle;
+    /* Build paths and evaluate their weights. Tracks the best weight.
+     * weightedSteerVecs holds all of the random steering samples and their
+     * path weights.
+     */
+    vector<WeightedSteeringVec> weightedSteerVecs(PATH_ITERATIONS);
+    double angle;
+    double bestWeight = 0;
     int increments = PATH_STAGE_TIME / TIME_INCREMENT;
-
     for(int i = 0; i < PATH_ITERATIONS; i++) {
         vector<double> steerPath(increments * PATH_STAGES, 0);
+        WeightedSteeringVec wsv;
         for(int s = 0; s < PATH_STAGES; s++) {
             angle = steeringSample();
+            wsv.steers.push_back(angle);
             fill_n(steerPath.begin() + s*increments, increments, angle);
         }
 
         sim_path sp = calculatePath(steerPath);
-        cost = calculatePathCost(sp, kdtree);
-        weightTotal += 1.0 / cost;
-
-        for(int j = 0; j < sp.angles.size(); j++) {
-            if(j < weightedSteeringRaw.size()) {
-                weightedSteeringRaw[j] += sp.angles[j] / cost;
-            } else {
-                weightedSteeringRaw.push_back(sp.angles[j] / cost);
+        wsv.weight = 1.0 / calculatePathCost(sp, kdtree);
+        weightedSteerVecs.push_back(wsv);
+
+        if(wsv.weight > bestWeight) bestWeight = wsv.weight;
+    }
+    //ROS_INFO("best weight %f", bestWeight);
+
+    /* Filter paths by weight and sort them by first steering value.
+     * Store the results in weightedSteerVecsFiltered.
+     */
+    vector<WeightedSteeringVec> weightedSteerVecsFiltered;
+    for(int i = 0; i < weightedSteerVecs.size(); i++) {
+        if(weightedSteerVecs[i].weight > bestWeight * ALT_PATH_THRESHOLD) {
+            weightedSteerVecsFiltered.push_back(weightedSteerVecs[i]);
+        }
+    }
+    //sort(weightedSteerVecsFiltered.begin(), weightedSteerVecsFiltered.end(), steeringVecCompare);
+
+    /* connect components based on distance. I refer to any random
+     * path sample as a steerVec here. The variable groups holds the
+     * sets of connected (connected = "close enough") paths
+     */
+    vector<SteeringGroup> groups;
+    // outer loop: iterate through good enough paths (around 10% of paths,
+    // depending on circumstances)
+    for(int i = 0; i < weightedSteerVecsFiltered.size(); i++) {
+        const WeightedSteeringVec &thisSteerVec = weightedSteerVecsFiltered[i];
+        bool foundMatch = false;
+        int thisGroupIndex;
```

Comment:
Try to avoid this* and that* variable names. Prefer names that better describe the functional or semantic difference between the two variables.